src/main/java/org/example/HomeworkException.java

```diff
@@ -0,0 +1,7 @@
+package org.example;
+
+public class HomeworkException extends Exception {
```

Comment:
remove this exception as it is redundant. 
For the future: DO NOT EXTEND from `Exception`! If you wanna create your custom exception, then extend from `RuntimeException` so a compiler won't warn you every time you use this custom exception in your code.